scipy/spatial/distance.py

```diff
@@ -1217,20 +1217,22 @@ def pdist(X, metric='euclidean', p=2, w=None, V=None, VI=None):
         if metric == minkowski:
             def dfun(u, v):
                 return minkowski(u, v, p)
+            X = _convert_to_double(X)
```

Comment:
Hmm, in that case I expect that other built in metrics will also need this conversion.